scripts/zkai_may2026_competitor_metric_matrix_gate.py

```
@@ -0,0 +1,651 @@
+    temps: list[pathlib.Path] = []
+    replaced: list[pathlib.Path] = []
+    original_bytes: dict[pathlib.Path, bytes | None] = {}
+    write_error: OSError | None = None
+    rollback_errors: list[str] = []
+
+    def write_temp(path: pathlib.Path, text: str) -> pathlib.Path:
+        _assert_no_repo_symlink_components(path.parent, "output path")
+        path.parent.mkdir(parents=True, exist_ok=True)
+        _assert_no_repo_symlink_components(path.parent, "output path")
+        if path.is_symlink():
+            raise CompetitorMetricMatrixError("output path must not include symlink components")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)